packages/apptive_grid_core/lib/src/model/form/form_block.dart

```diff
@@ -0,0 +1,78 @@
+/// A freeform block inside of a form
+class FormBlock {
+  /// Deserializes [json] into a FormBlock Object
+  factory FormBlock.fromJson(dynamic json) => FormBlock(
+        id: json['id'],
+        text: json['text'],
+        fieldIndex: json['fieldIndex'],
+        type: json['type'],
+        style: json['style'],
+        pageId: json['pageId'],
+      );
+
+  /// Creates a FormBlock Object
+  FormBlock({
+    required this.id,
+    required this.pageId,
+    required this.fieldIndex,
+    required this.text,
+    required this.type,
+    required this.style,
+  });
+
+  /// Id of the block
+  final String id;
+
+  /// Page of the block in the form
+  final String pageId;
+
+  /// Position of the block in the page
+  final int fieldIndex;
```

Comment:
Is this really a "field" index?

Reply:
This is the backend name, but I agree, that it makes not much sense outside of the form fields. I renamed the attribute in both to positionOnPage